### PR TITLE
support TEQ method.

### PR DESCRIPTION
**Pull request by lkk12014402**
## Type of Change

Add feature that support Trainable Equivalent Transformation (TEQ)
Original API not changed 

## Description
Trainable Equivalent Transformation

### CODE REVIEW

File: neural_compressor/adaptor/torch_utils/teq.py [hunk 1]
```diff
@@ -0,0 +1,485 @@
...
+            ctx: Pytorch convention.
+            inputs: A Tensor of type float32.
+            min_range: A float.
+            max_range: A float.
+            num_bits: An integer
+
+        Returns:
+            outputs: A Tensor of type output_dtype
+        """
+        return quant_weight(inputs, num_bits, group_size)
```
Comment (wenhuach21) [inline neural_compressor/adaptor/torch_utils/teq.py:56]: need to support schema

File: neural_compressor/adaptor/torch_utils/teq.py [hunk 2]
```diff
@@ -0,0 +1,485 @@
...
+    def train(self, train_steps=100, lr=1e-3, warmup_ratio=0.05, gradient_accumulation_steps=1, logging_steps=10,
+            betas=[0.9, 0.9], weight_decay=0, optimizer=None, lr_scheduler=None, lr_scheduler_type="linear"):
+        """
+        train function
+        """
+        if optimizer is None:
+            trained_alphas_list = []
+            for item in self.trained_alphas.items():
+                trained_alphas_list.append(item[1])
+            optimizer = torch.optim.Adam(trained_alphas_list, lr=lr, weight_decay=weight_decay, betas=betas)
```
Comment (wenhuach21) [inline neural_compressor/adaptor/torch_utils/teq.py:411]: AdamW, if weight_decay is not 0, there are different.

File: neural_compressor/adaptor/torch_utils/teq.py [hunk 3]
```diff
@@ -0,0 +1,485 @@
...
+        while True:
+            for inputs in self.dataloader:
+                if isinstance(inputs, dict):
+                    input_id = inputs["input_ids"]
+                else:
+                    input_id = inputs[0]
+
+                input_id = input_id.to(self.device)
+                output = self.model(input_id, labels=input_id)
+                loss = output[0] / gradient_accumulation_steps
```
Comment (wenhuach21) [inline neural_compressor/adaptor/torch_utils/teq.py:435]: need to support customized loss function later

File: neural_compressor/adaptor/torch_utils/teq.py (hunk 2 above)
Comment (lkk12014402) [reply neural_compressor/adaptor/torch_utils/teq.py:411]: train func will be moved outside for custom use.

File: neural_compressor/adaptor/torch_utils/teq.py (hunk 3 above)
Comment (lkk12014402) [reply neural_compressor/adaptor/torch_utils/teq.py:435]: train func will be moved outside for custom use.

Comment (wenhuach21) [reply neural_compressor/adaptor/torch_utils/teq.py:435]: better have a default one in the code
